### ISSUE TITLE
MCOIMAPSession not working for godaddy mail server(pop.secureserver.net) 

### ISSUE THREAD

**Issue by iChirag** [MailCore/mailcore2#1101]
I had used Mailcore library to receive and sent email on one my iOS application. However,I got success it to use with popular email server like Gmail,Ymail and Hotmail.

Now I am trying to use that library to fetch emails from my godaddy mail server [pop.secureserver.net] or other email server.

This is my  code:

MCOIMAPSession *session = [[MCOIMAPSession alloc] init];
    [session setHostname:@"pop.secureserver.net"];
    [session setPort:993];
    [session setUsername:self.txtEmail.text];
    [session setPassword:self.txtPassword.text];
    [session setConnectionType:MCOConnectionTypeTLS];//I also try MCOConnectionTypeStartTLS
    [session setCheckCertificateEnabled:NO];

 MCOIMAPMessagesRequestKind requestKind = MCOIMAPMessagesRequestKindHeaders|MCOIMAPMessagesRequestKindFlags;

    NSString *folder = @"INBOX";
    MCOIndexSet *uids = [MCOIndexSet indexSetWithRange:MCORangeMake(1, UINT64_MAX)];

    MCOIMAPFetchMessagesOperation *fetchOperation = [session fetchMessagesOperationWithFolder:folder requestKind:requestKind uids:uids];
    //    [SVProgressHUD show];

    [fetchOperation start:^(NSError * error, NSArray * fetchedMessages, MCOIndexSet * vanishedMessages) {
        //Let's check if there was an error:
        if(error) {
            NSLog(@"Error downloading message headers:%@", error);

        }

    }];

I am getting Error as below :

Error downloading message headers:Error Domain=MCOErrorDomain Code=1 "A stable connection to the server could not be established." UserInfo=0x7fa6a1e3cdc0 {NSLocalizedDescription=A stable connection to the server could not be established.}

Am I missing something in code or settings?

Please help me to resolve this.

**Comment by dinhviethoa** [comment]
Please read the instructions here:
https://support.godaddy.com/help/article/4714/setting-up-your-email-address-with-imap
It's imap.secureserver.net

**Comment by iChirag** [comment]
hi,

Thanks for update.

Actually I am facing issue to connect to mail server from hyperpma.com

This is my server detail 
MCOIMAPSession *session = [[MCOIMAPSession alloc] init];
[session setHostname:@"host.hyperpma.info"];
[session setPort:993];
[session setUsername:self.txtEmail.text];
[session setPassword:self.txtPassword.text];
[session setConnectionType:MCOConnectionTypeTLS];//I also try MCOConnectionTypeStartTLS
[session setCheckCertificateEnabled:NO];

![screen shot 2015-04-03 at 10 05 11 pm](https://cloud.githubusercontent.com/assets/3703405/6985132/a175cdbc-da4d-11e4-911e-cd7485054218.png)

Unfortunately,I gave here Godaddy webmail server because it little famous.I thought if it resolved then I will resolve my issue as well.
I got solution for godaddy still have issue for hyperpma.com server.

Still I got same error.

Error downloading message headers:Error Domain=MCOErrorDomain Code=1 "A stable connection to the server could not be established." UserInfo=0x7fa6a1e3cdc0 {NSLocalizedDescription=A stable connection to the server could not be established.}


Thanks.




I had attached screenshot for server detail as well.
